### PR TITLE
refactor(#2389): Renamed classes `OnDefault` and `OnVersioned`

### PR DESCRIPTION
**Pull request by maxonfjvipon**
Closes: #2389

---

## PR-Codex overview
This PR focuses on replacing the usage of the `OnDefault` class with the `OnVersioned` and `OnReplaced` classes in the `eo-maven-plugin` module. 

### Detailed summary
- Replaced `OnDefault` with `OnVersioned` and `OnReplaced` in multiple files.
- Updated imports and references to the new classes.
- Updated test cases to use the new classes.

> The following files were skipped due to too many changes: `eo-maven-plugin/src/main/java/org/eolang/maven/name/OnReplaced.java`, `eo-maven-plugin/src/test/java/org/eolang/maven/name/OnReplacedTest.java`, `eo-maven-plugin/src/main/java/org/eolang/maven/name/OnVersioned.java`, `eo-maven-plugin/src/test/java/org/eolang/maven/name/OnVersionedTest.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo please review this one

File: eo-maven-plugin/src/main/java/org/eolang/maven/name/OnVersioned.java [hunk 1]
```diff
@@ -23,76 +23,64 @@
...
+ * @since 0.29.6
  */
 public final class OnVersioned implements ObjectName {
 
     /**
-     * Delimiter between name and hash in EO object name.
+     * Object with or without hash.
      */
-    public static final String DELIMITER = "#";
+    private final Unchecked<String> object;
```
Comment (volodya-lombrozo) [inline eo-maven-plugin/src/main/java/org/eolang/maven/name/OnVersioned.java:40]: @maxonfjvipon Why do we need to use `Unchecked` here? Can we keep the simple code with `String`?

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]

File: eo-maven-plugin/src/main/java/org/eolang/maven/ProbeMojo.java [hunk 2]
```diff
@@ -166,8 +166,8 @@ private Collection<ObjectName> probes(final Path file) throws FileNotFoundExcept
                 obj -> new OnCached(
                     new OnSwap(
                         this.withVersions,
-                        new OnDefault(
-                            new OnVersioned(ProbeMojo.noPrefix(obj), this.hashes),
+                        new OnVersioned(
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/main/java/org/eolang/maven/ProbeMojo.java:169]: @volodya-lombrozo here

File: eo-maven-plugin/src/main/java/org/eolang/maven/name/OnVersioned.java (hunk 1 above)
Comment (maxonfjvipon) [reply eo-maven-plugin/src/main/java/org/eolang/maven/name/OnVersioned.java:40]: @volodya-lombrozo this change was added [here](https://github.com/objectionary/eo/pull/2386), so it's not a part of the PR. But there's a [reason](https://github.com/objectionary/eo/pull/2391/files/716599c381c2dad36a8b26dc9418bdcd8ddbd434#r1294305053) for using `Unchecked`

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/name/OnVersioned.java:40]: @maxonfjvipon I see, thanks.

**Review by volodya-lombrozo — APPROVED** [review]
@maxonfjvipon Looks good to me. Thank you.

**Comment by maxonfjvipon** [comment]
@yegor256 please have a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**